NonsoAmadi10/Auto-Mart
# #166730145 Creates User should be able to post a Car advert

### What does this PR do??
- Creates ` POST /api/v1/car `

### Description of Task Taken to complete??
- Uncomment test cases
- Run `npm test ` to see test fail
- Implement feature
- Protect route with JSON web token
- Validate input request

- How should this be manually Tested??
- Clone this repo
- ` git checkout ft-user-should-post-ad-166730145 `
-   Run ` npm install ` in the terminal
- Setup all environment variables in the .env.example file
- Run ` npm run dev-start` to start this server
- Open postman on ` POST /api/v1/car `
- Enter the following in the request body :
``` 
{
'photo': a Valid photograph file,
'manufacturer': String,
'model': String,
'state': String,
bodyType: String,
'price': Float
}
```

### Any Background Context you want to provide??
- You need to obtain a token to use this feature. see #50 to register and #51 to get a token
- The request body type must be  multipart/form-data and not a JSON body


### What are the relevant pivotal tracker stories??
[166730145](https://www.pivotaltracker.com/story/show/166730145)
